Further modify bucket logic for releases/internal bucket

amd-chrissosa opened the pull request:
## Motivation

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->

## Technical Details

<!-- Explain the changes along with any relevant GitHub links. -->

## Test Plan

<!-- Explain any relevant testing done to verify this PR. -->

## Test Result

<!-- Briefly summarize test outcomes. -->

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

Commit:
Further modify bucket logic for releases/internal bucket

amd-justchen reviewed (commented):
I only understand a bit of the context for this change but the code seems good to me! logic for no `external_repo` is the same as the original check while adding internal and external buckets.

Waiting on workflow success.

umarinkovic committed:
updated releases.md to include [product name] -> [gfx family] mapping… (#1328)

## Motivation

Resolved a TODO in RELEASES.md

---------

Co-authored-by: Marius Brehler <marius.brehler@gmail.com>

ScottTodd committed:
[doc] Add build architecture diagrams to development guide. (#1311)

## Motivation

Progress on https://github.com/ROCm/TheRock/issues/1028, adding more
diagrams to help introduce "build from source guide" concepts.

Next, I think we can expand on the "Build using CMake" part of the
diagram with new diagrams and accompanying text showing the various
options there for
* from-scratch source builds
* bootstrapped builds
* enabling/disabling prebuilt markers for individual subprojects
* builds from within each subproject
* etc.

## Technical Details

The new diagrams were authored in https://excalidraw.com/. The svgs
contain the scene information so they can be loaded back into the
editor.

---------

Co-authored-by: Marius Brehler <marius.brehler@gmail.com>

ScottTodd committed:
[doc] Add subproject build diagrams to development guide. (#1324)

## Motivation

Progress on https://github.com/ROCm/TheRock/issues/1028. This adds
diagrams for how subproject builds connect to one another via build
artifacts. This introduces concepts that are used by the
[`buildctl.py`](https://github.com/ROCm/TheRock/blob/main/build_tools/buildctl.py)
and
[`bootstrap_build.py`](https://github.com/ROCm/TheRock/blob/main/build_tools/bootstrap_build.py)
scripts to enable productive workflows on individual components.

## Technical Details

More new diagrams authored in https://excalidraw.com/. The svgs contain
the scene information so they can be loaded back into the editor.

I've also exported dark mode versions of all the new diagrams and
removed the icons from the "build architecture" diagram (icons did not
handle the dark mode color inversion well).

marbre committed:
Switch to build from rocm-systems (#1325)

This switches to build
* base/rocm-core
* base/rocm_smi_lib
* base/rocprofiler-register
* core/HIP
* core/ROCR-Runtime
* core/clr
* core/rocminfo
* profiler/aqlprofiler
* profiler/rocprofiler-sdk
* profiler/roctracer

from rocm-systems instead from individual submodules:
* Adds the rocm-systems submodule
* Removes no longer needed submodules
* Rebases patches from the individual projects to rocm-systems
* Updates the helper scripts in `build_tools`
* Bumps core/amdgpu-windows-interop

---------

Co-authored-by: JC <justchen@amd.com>

geomin12 committed:
[ci] Adding post build script (#1295)

After discussion and revert from #1291 , we decided to create a post
build script to help prevent indirection for developers.

This script combines previous upload scripts into one script, with
options for CI and upload.

marbre committed:
Fix setting external source dirs (#1333)

The `THEROCK_${_PACKAGE}_SOURCE_DIR` variables were not correctly set
and propagated.

amd-justchen reviewed (approved):
Following up on 
https://github.com/ROCm/TheRock/pull/1316#pullrequestreview-3153294796

Looks like the workflows passed!

marbre commented:
> Following up on [#1316 (review)](https://github.com/ROCm/TheRock/pull/1316#pullrequestreview-3153294796)
> 
> Looks like the workflows passed!

This needs a rebase first as https://github.com/ROCm/TheRock/commit/21711555f7f3468b7dff45f176b47523bb55dca6 landed.

marbre committed:
Bump rocm-systems submodule 20250827 (#1335)

marbre committed:
Bump rccl submodule 20250827 (#1336)

umarinkovic committed:
[doc] Added a Target Family Selection section to python_packaging.md (#1338)

## Motivation

Updated python_packaging.md to reflect the changes added in:
https://github.com/ROCm/TheRock/pull/1269

ScottTodd committed:
Bump DLL versions from 0700 to 0701 after rocm-systems update. (#1341)

## Motivation

Fixes https://github.com/ROCm/TheRock/issues/1340.

## Technical Details

We still hardcode library names
(https://github.com/ROCm/TheRock/issues/1057). We might want to land
https://github.com/ROCm/TheRock/pull/1061 in some form to change that.

## Test Plan

I checked the DLL names in my local build directory. This exposed more
gaps in our CI/CD test coverage that should have blocked either
https://github.com/ROCm/TheRock/commit/0c23e81041c9bd839c5d2c4cf7bc28ab69bf653a
or publishing the Python packages.

I also modified
`D:\scratch\therock\3.12.venv\Lib\site-packages\rocm_sdk\_dist_info.py`
and ran `rocm-sdk test` successfully.

I could also build the Python packages locally to verify.

## Test Result

<!-- Briefly summarize test outcomes. -->

## Submission Checklist

- [x] Look over the contributing guidelines at
https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

ScottTodd committed:
Drop amdgpu-windows-interop checkout from release_windows_packages.yml. (#1342)

Follow-up to https://github.com/ROCm/TheRock/pull/887 where this was
missed. The project is now included in its own submodule, no extra
checkout needed.

ScottTodd committed:
[ci] Rename xfail matrix to nightly, mark 90x and 101x as passing. (#1343)

## Motivation

Progress on https://github.com/ROCm/TheRock/issues/1337. Related to
https://github.com/ROCm/TheRock/issues/1097.

## Technical Details

gfx90X-dcgpu and gfx101X-dgpu builds have been passing on nightly runs
on
https://github.com/ROCm/TheRock/actions/workflows/ci_nightly.yml?query=branch%3Amain

This allows us to separate the _policy decision_ "what support level is
a given gfx family" from the _technical reality_ "does that gfx family
pass builds/tests". Now we can have the nightly job include gfx families
that are passing builds/tests instead of only having "xfail" jobs there.

While I was renaming the matrix variables I also did related refactoring
to hopefully make this code easier to maintain going forward.

## Test Plan

* Ran `python .\build_tools\github_actions\tests\configure_ci_test.py`
locally
* Confirm which targets are selected to run on this PR
* Watch which targets are selected on postsubmit
* Watch which targets are selected on the next run on
https://github.com/ROCm/TheRock/actions/workflows/ci_nightly.yml?query=branch%3Amain

## Test Result

* Local unit tests passed

## Submission Checklist

- [x] Look over the contributing guidelines at
https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

araravik-psd committed:
Add gating mechanism to ensure PyTorch wheels pass tests before releasing (#1110)

This PR introduces a gating mechanism to ensure PyTorch wheels are
tested before being published to the release bucket.

Implementation design and alternatives were explored and documented in
#1072
Only the solution below was selected and implemented:


-------------------------------------------------------------------------------------------------

1. Build PyTorch wheels
2. Upload the built wheels to the v2-staging (staging bucket)
3. Run PyTorch tests using wheels from the staging bucket
4. Only if tests pass or if there is an override for uploading wheels without test(using bypass_tests_for_releases from amdgpu_family_matrix.py), copy the validated wheels from the staging bucket to the release bucket


----------------------------------------------------------------------------------------------------
This ensures that only tested and verified wheels are published to the
release index.

---------

Co-authored-by: arravikum <arravikum@amd.com>

geomin12 committed:
[ci] Fixing forked TheRock testing (#1339)

Closes #1334 

Silly mistake from me, looks like sanity check was missing a flag.

I'm planning to combine this into one script/workflow and work on
various cleanup agendas, so this won't happen in the future!

marbre committed:
Temporarily deactivate scheduled Windows releases (#1353)

Stops scheduled releases until #1347 is fixed.

kzhuravl committed:
Add spirv-llvm-translator submodule (#1330)

- Initially pointing to amd-mainline branch

ScottTodd committed:
Warn about unstable Windows builds while broken by the rocm-systems migration. (#1355)

See https://github.com/ROCm/TheRock/issues/1347.

The CMake configure warning looks like this in context:
```
[cmake] -- Configuring background job pool for 6 concurrent jobs
[cmake] CMake Warning at CMakeLists.txt:35 (message):
[cmake]   *************************************************
[cmake] 
[cmake]   Windows builds are currently unstable during the migration to rocm-systems!
[cmake]   Builds are expected to succeed but there are runtime errors and timeouts.
[cmake]   The last known good commit on Windows is
[cmake]   https://github.com/ROCm/TheRock/commit/529fa36d69c7d4295317a70d3c5c88000a364b2c
[cmake]   See https://github.com/ROCm/TheRock/issues/1347 for details.
[cmake] 
[cmake]   *************************************************
[cmake] 
[cmake] 
[cmake] -- Enabling building tests
[cmake] -- ROCm version: 7.0.0
[cmake] -- HIP version: 7.1.0
[cmake] -- Implicitly enabled features: HOST_BLAS
```

ScottTodd committed:
Switch hip-tests (for hipInfo) to rocm-systems subfolder. (#1352)

Fixes https://github.com/ROCm/TheRock/issues/1351.

We may want a full project for hip-tests instead of just extracting
hipInfo from it. That's for later.

raramakr committed:
Native packaging in TheRock (#1252)

Create RPM and DEB package from Rock artifacts

## Motivation

<!-- To create RPM and DEB packages from Rock artifacts. -->

## Technical Details

<!-- Required package details are listed in json file. The scripts will
download the corresponding Rock artifacts and create rpm/deb packages
based on the package details in json . -->

## Test Plan

<!--Executed the scripts in UB-24.04 machine. -->

## Test Result

<!-- RPM/DEB packages are getting created for the packages listed in
json file . -->

## Submission Checklist

- [ ] Look over the contributing guidelines at
https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

---------

Co-authored-by: raramakr <raramakr@amd.com>

araravik-psd committed:
Adding attributes to release workflows (#1357)

Add inputs added for build_portable_linux_pytorch_wheels.yml from #1110 

Added required inputs for both the release_portable_linux_packages(Which
runs build_portable_linux_pytorch_wheels.yml) and also
release_portable_linux_pytorch_wheels.

---------

Co-authored-by: arravikum <arravikum@amd.com>

ScottTodd committed:
Fix Windows on new rocm-systems with some revert patches. (#1356)

## Motivation

Progress on https://github.com/ROCm/TheRock/issues/1347.

## Technical Details

1. Patch 10 is a revert of
https://github.com/ROCm/rocm-systems/commit/72b9408fedb08da67d0f3fec89316fb6e44153f2,
with conflicts manually resolved. Something in that change causes
programs (including many of our math-libs tests) to hang. A fix-forward
is also up for review at https://github.com/ROCm/rocm-systems/pull/809.
2. Patch 11 is essentially a cherry-pick of
https://github.com/ROCm/rocm-systems/pull/802. This commit
https://github.com/ROCm/rocm-systems/commit/2ff2316227dc3bb04447060c719e78baf2d7383f
ran clang-format on
https://github.com/ROCm/rocm-systems/blob/develop/projects/clr/rocclr/device/pal/palblitcl.cpp,
resulting in this broken file: [`palblitcl.cpp` @
df11d9dc](https://github.com/ROCm/rocm-systems/blob/df11d9dcc21dcdd8446d690d8bf0cceca9b14268/projects/clr/rocclr/device/pal/palblitcl.cpp)
instead of this working file: [`palblitcl.cpp` @
fe14cc34](https://github.com/ROCm/rocm-systems/blob/fe14cc349baaab4aa79534296bdcef0b5957ee77/projects/clr/rocclr/device/pal/palblitcl.cpp).
That resulted in errors like this when running programs like
`hipInfo.exe`:
    
    ```
C:\Users\NOD-SH~1\AppData\Local\Temp\comgr-505531\input\Assembly
Text:252:1: error: symbol 'trap_entry' is already defined
    trap_entry :
    ^
C:\Users\NOD-SH~1\AppData\Local\Temp\comgr-505531\input\Assembly
Text:281:1: error: symbol '.not_abort_trap' is already defined
    .not_abort_trap :
    ^
    ```

## Test Plan

* Sanity check with some local builds
* Opt-in to gfx11X-windows tests on presubmit and watch CI

## Test Result

Local builds on gfx1100:

* CMake build completed
  * `hipInfo` works as expected
  * `hipblaslt-test '--gtest_filter=*pre_checkin*'` is passing
* ROCm python package build completed
  * `rocm-sdk test` works as expected
* PyTorch build completed
  * `pytest -v external-builds/pytorch/smoke-tests` passed

All CI workflows for gfx1151 that we expect to pass are passing.

## Submission Checklist

- [x] Look over the contributing guidelines at
https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

Commit:
Rebase on top of Geo's move of this logic.

Commit:
Delete it

amd-chrissosa commented:
Agh, I thought I committed this. Spent way too long trying to rebase but giving up. Superseding this PR with: https://github.com/ROCm/TheRock/pull/1365
